src/test/java/org/takes/rs/RsPrettyXMLTest.java

```diff
@@ -51,7 +51,60 @@ public void formatsXmlBody() throws IOException {
                     new RsWithBody("<test><a>foo</a></test>")
                 )
             ).printBody(),
-            Matchers.is("<test>\n   <a>foo</a>\n</test>")
+            Matchers.is("<test>\n   <a>foo</a>\n</test>\n")
+        );
+    }
+
+    /**
+     * RsPrettyXML can format HTML5 markup with proper DOCTYPE.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    // @checkstyle MethodNameCheck (1 line)
+    public void formatsHtml5DoctypeBody() throws IOException {
+        MatcherAssert.assertThat(
+            new RsPrint(
+                new RsPrettyXML(
+                    new RsWithBody(
+                        "<!DOCTYPE html><html><head></head><body></body></html>"
+                    )
+                )
+            ).printBody(),
+            Matchers.containsString("<!DOCTYPE HTML>")
+        );
+    }
+
+    /**
+     * RsPrettyXML can format HTML4 markup with DOCTYPE with public
+     * and system id.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    // @checkstyle MultipleStringLiteralsCheck (25 line)
```

Comment:
@dimonn12 try to join in a smart way and also use local variables if necessary to get rid of this.